test/e2e/operator/nmstate_install_test.go

```diff
@@ -0,0 +1,123 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1beta1"
+)
+
+func installNMState(nmstate nmstatev1beta1.NMState, nmstateKey types.NamespacedName) {
+	_, err := getNMStateOperator()
+	Expect(apierrors.IsNotFound(err)).To(BeTrue(), "NMState CR does not exist yet")
+	err = framework.Global.Client.Create(context.TODO(), &nmstate, &framework.CleanupOptions{})
+	Expect(err).ToNot(HaveOccurred(), "NMState CR created without error")
+}
+
+func uninstallNMState(nmstate nmstatev1beta1.NMState, nmstateKey types.NamespacedName) {
+	_, err := getNMStateOperator()
+	Expect(apierrors.IsNotFound(err)).To(BeFalse(), "NMState CR exists")
+	err = framework.Global.Client.Delete(context.TODO(), &nmstate, &client.DeleteOptions{})
+	Expect(err).ToNot(HaveOccurred(), "NMState CR successfully removed")
+}
+
+func getNMStateCRKey() types.NamespacedName {
+	return types.NamespacedName{Namespace: "nmstate", Name: "nmstate"}
+}
+
+func getNMStateOperator() (nmstatev1beta1.NMState, error) {
+	var nmstate nmstatev1beta1.NMState
+	err := framework.Global.Client.Get(context.TODO(), getNMStateCRKey(), &nmstate)
+	return nmstate, err
+}
+
+func getHandlerKey() types.NamespacedName {
+	return types.NamespacedName{Namespace: "nmstate", Name: "nmstate-handler"}
+}
+
+func getNMStateHandler() (v1.DaemonSet, error) {
+	var ds v1.DaemonSet
+	err := framework.Global.Client.Get(context.TODO(), getHandlerKey(), &ds)
+	return ds, err
+}
+
+var _ = Describe("NMState operator", func() {
+	var (
+		nmstate    nmstatev1beta1.NMState
+		webhookKey = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-webhook"}
+	)
+	BeforeEach(func() {
+		nmstate = nmstatev1beta1.NMState{
```

Comment:
Could you set the NMStateCRKey here directly? Then we can git of the helper get function and make installNMState signature simpler, just pass it nmstate.